--- ext/standard/mail.c	2016-03-29 16:44:19.000000000 +0200
+++ ext/standard/mail.c	2016-04-12 17:04:37.639669444 +0200
@@ -319,15 +319,22 @@
 	if (PG(mail_x_header)) {
 		const char *tmp = zend_get_executed_filename();
 		zend_string *f;
+		long script_inode;
 
 		f = php_basename(tmp, strlen(tmp), NULL, 0);
 
+		/* Getting the executed script inode */
+		php_stat(tmp, strlen(tmp), FS_INODE, &script_inode);
+
 		if (headers != NULL && *headers) {
 			spprintf(&hdr, 0, "X-PHP-Originating-Script: " ZEND_LONG_FMT ":%s\n%s", php_getuid(), ZSTR_VAL(f), headers);
 		} else {
 			spprintf(&hdr, 0, "X-PHP-Originating-Script: " ZEND_LONG_FMT ":%s", php_getuid(), ZSTR_VAL(f));
 		}
 		zend_string_release(f);
+
+		/* Adding a header containing the script inode */
+		spprintf(&hdr, 0, "%s\nX-PHP-Originating-Script-Inode: %d", hdr, script_inode);
 	}
 
 	if (hdr && php_mail_detect_multiple_crlf(hdr)) {
